mscore/exportmidi.cpp

```diff
@@ -79,6 +79,30 @@ void ExportMidi::writeHeader()
             }
 #endif
 
+      //--------------------------------------------
+      //    write track and instrument names
+      //--------------------------------------------
+
+      int staffIdx = 0;
+      for (auto& track1: mf.tracks()) {
+            Staff* staff  = cs->staff(staffIdx);
+
+            int len = staff->partName().length() + 1;
+            unsigned char* data = new unsigned char[len];
+
+            memcpy(data, staff->partName().toLatin1().data(), len);
```

Comment:
Although the MIDI specification does not mandate a specific character encoding, some software is able to handle UTF-8-encoded strings.

I created a test score that contained a “E♭ Bass” instrument, exported it with your patch, and opened it in [REAPER](https://www.reaper.fm/). The flat symbol was lost and replaced and the track name showed up as “E? Bass”. I then modified your code to export in UTF-8 and repeated the test, and the track name showed up perfectly in REAPER.

So I recommend that you use UTF-8 instead of the current Latin 1 (ISO/IEC 8859-1). Note that if you do so, you will have to redo the length calculation, since UTF-8 is a variable-length encoding.

Reply:
I couldn't find anything regarding character encoding in the [MIDI specs](http://www.music.mcgill.ca/~ich/classes/mumt306/StandardMIDIfileformat.html) so ASCII was the best choice. But I had the same problem importng B♭ Clarinet in Rosegarden.
I'll see whether UTF-8 will cause any issues.